packages/schematics/angular/guard/index.ts

```diff
@@ -7,14 +7,35 @@
  */
 
 import { Rule, SchematicsException } from '@angular-devkit/schematics';
+
 import { generateFromFiles } from '../utility/generate-from-files';
+
 import { Implement as GuardInterface, Schema as GuardOptions } from './schema';
 
 export default function (options: GuardOptions): Rule {
-  if (!options.implements) {
-    throw new SchematicsException('Option "implements" is required.');
+  if (!options.implements && !options.guardType) {
+    throw new SchematicsException('Option "implements" or "guardType" is required.');
+  }
+  if (options.implements && options.implements.length > 0 && options.guardType) {
+    throw new SchematicsException('Options "implements" and "guardType" cannot be used together.');
+  }
+
+  return options.guardType ? functionalGuardRule(options) : classBasedGuardRule(options);
+}
+
+function functionalGuardRule(options: GuardOptions): Rule {
+  if (!options.guardType) {
+    throw new SchematicsException('Option "implements" or "guardType" is required.');
```

Comment:
Copy/paste mistake?

Reply:
Typescript isn't quite able to make this work. I've removed the helper functions.